website/docs/attached-class.md

```diff
@@ -0,0 +1,75 @@
+---
+id: attached-class
+title: T.attached_class
+---
+
+`T.attached_class` can be used to refer to the type of instances from a
+singleton-context, in a way that is friendly to inheritance.
+
+```ruby
+class AbstractData
+  extend T::Sig
+
+  sig {params(id: Integer).returns(T.attached_class)}
+  def self.load(id)
+    new
+  end
+end
+
+class Article < AbstractData; end
+class User < AbstractData; end
+class Administrator < User; end
+
+T.reveal_type(Article.load(10)) # Article
+T.reveal_type(Administrator.load(20)) # Administrator
+```
+
+# Typing Factory Methods
+
+Consider the following example of a class hierarchy with a factory method:
+
+```ruby
+class Parent
+  extend T::Sig
+
+  sig {returns(Parent)}
```

Comment:
Can you add a comment on this line that indicates that there's something subtly wrong with it, and that it'll be fixed in a future example? Like:

```ruby
# "correct" but not ideal:
```

or something of your choosing.

People skimming reference pages don't usually read the surrounding context to figure out whether this is an example they shouldn't copy.